src/Api/Controllers/OrganizationExportController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Bit.Api.Models.Response;
+using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Settings;
+using Core.Models.Data;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Bit.Api.Controllers
+{
+    [Route("organizations/{organizationId}")]
+    [Authorize("Application")]
+    public class OrganizationExportController : Controller
+    {
+        private readonly ICipherRepository _cipherRepository;
+        private readonly ICollectionCipherRepository _collectionCipherRepository;
+        private readonly ICollectionRepository _collectionRepository;
+        private readonly ICurrentContext _currentContext;
+        private readonly IProviderService _providerService;
+        private readonly IUserService _userService;
+        private readonly GlobalSettings _globalSettings;
+
+        public OrganizationExportController(
+            ICipherRepository cipherRepository,
+            ICollectionCipherRepository collectionCipherRepository,
+            ICollectionRepository collectionRepository,
+            ICurrentContext currentContext,
+            IProviderService providerService,
+            IUserService userService,
+            GlobalSettings globalSettings)
+        {
+            _cipherRepository = cipherRepository;
+            _collectionCipherRepository = collectionCipherRepository;
+            _collectionRepository = collectionRepository;
+            _currentContext = currentContext;
+            _providerService = providerService;
+            _userService = userService;
+            _globalSettings = globalSettings;
+        }
+
+        [HttpGet("export")]
+        public async Task<OrganizationExportResponseModel> Export(Guid organizationId)
```

Comment:
There are several areas within the methods of this controller that have repeated code from the individual implementations. Is there a way we can consolidate some of the functionality from the controller layer to the service layers and reuse them here instead of repeating ourselves?

Reply:
That's a good idea, I'll move the code to the CipherService and CollectionsService.

Reply:
Awesome!